posthog/clickhouse/client/execute.py

```diff
@@ -108,12 +110,19 @@ def sync_execute(
             )
         except Exception as err:
             err = wrap_query_error(err)
-            statsd.incr(
-                "clickhouse_sync_execution_failure",
-                tags={"failed": True, "reason": type(err).__name__},
-            )
-
-            raise err
+            if type(err).__name__ == "CHQueryErrorQueryWasCancelled":
+                statsd.incr("clickhouse_execution_cancelled", tags={"team_id": team_id})
+            elif type(err).__name__ == "EstimatedQueryExecutionTimeTooLong":
+                statsd.incr(
+                    "clickhouse_execution_timeout",
+                    tags={"team_id": team_id},
+                )
+            else:
+                statsd.incr(
+                    "clickhouse_sync_execution_failure",
+                    tags={"failed": True, "reason": type(err).__name__},
+                )
+                raise err
```

Comment:
In the branches that don't raise we'd try to return `return` when it's not bound, which would fail badly. We need to still raise. And we can't return None or an empty array, because a lot of call sites will expect something in the format of their SELECT – so they would fail with a different return type

Reply:
Yeah, definitely need to rethink this 🤔